docs/installation.md

```diff
@@ -0,0 +1,106 @@
+# Installation guide for ClusterLink
+The ClusterLink gateway contains three main components: the control-plane, data-plane, and gwctl (more details can be found [here](../README.md#what-is-clusterlink)).
+
+ClusterLink can be deployed in any K8s based cluster (e.g., google GKE, amazon EKS, IBM IKS, KIND etc.).
+
+To deploy the ClusterLink gateway in a K8s cluster, follow these steps:
+
+1. Create and deploy certificates and the ClusterLink deployment YAML file.
+2. Expose the ClusterLink deployment to a public IP.
+3. Optionally, create a central gwctl component to control one or more gateways.
+
+Before you begin, please build the project according to the [instructions](../README.md#building-clustelink).
+## 1. ClusterLink deployment and certificates
+In this step, we generate the ClusterLink certificates and deployment files.
+
+1) Create a folder (eg. DEPLOY_DIR) for all deployments and yaml files:
+
+        export PROJECT_DIR=`git rev-parse --show-toplevel`
+        export DEPLOY_DIR=~/clusterlink-deployment
+        mkdir -p $DEPLOY_DIR
+        cd $DEPLOY_DIR
+
+2) Create fabric certificate, which is the root CA certificate for all the clusters:
+
+        $PROJECT_DIR/bin/cl-adm create fabric
+
+    Note: This step needs to be done only once per fabric.
+3) Create certificates and a deployment file for the cluster (e.g., peer1):
+
+        $PROJECT_DIR/bin/cl-adm create peer --name peer1
+
+    Note: By default, the images are pulled from ghcr.io/clusterlink-net. To change the container registry, you can use the ```--container-registry``` flag.  
+    This step needs to be performed for every K8s cluster.
+
+4) Apply ClusterLink deployment:
+   
+        kubectl apply -f $DEPLOY_DIR/peer1/k8s.yaml
+
+5) Verify that all components (cl-controlplane, cl-dataplane, gwctl) are set up and ready.
+   
+        kubectl rollout status deployment cl-controlplane
+        kubectl rollout status deployment cl-dataplane
+        kubectl wait --for=condition=ready pod -l app=gwctl 
+        
+    Expected output:
+    
+        deployment "cl-controlplane" successfully rolled out
+        deployment "cl-dataplane" successfully rolled out
+        pod/gwctl condition met
+
+## 2. Expose the ClusterLink deployment to a public IP
+Create a public IP for accessing the ClusterLink gateway.
+For a testing environment (e.g., KIND), create a K8s nodeport service:
+
+        kubectl apply -f $PROJECT_DIR/demos/utils/manifests/kind/cl-svc.yaml
```

Comment:
I don't like the fact that the installation doc refers to some yaml from one of the demos.
Can we define cl-svc.yaml here?
e.g.
```
cat <<EOF >> cl-svc.yaml
apiVersion: v1
kind: Service
metadata:
  name: cl-svc
spec:
  type: NodePort
  selector:
    app: cl-dataplane
  ports:
    - port: 443
      targetPort: 443
      nodePort: 30443
      protocol: TCP
      name: http
EOF
```